nmagent/client.go

```diff
@@ -284,6 +288,70 @@ func (c *Client) GetHomeAz(ctx context.Context) (AzResponse, error) {
 	return homeAzResponse, nil
 }
 
+func (c *Client) RefreshSecondaryIPsIfNeeded(ctx context.Context) (refreshNeeded bool, ips []string, err error) {
+	if time.Since(c.secondaryIPLastRefreshTime) < c.secondaryIPQueryInterval {
+		return false, nil, nil
+	}
+
+	c.secondaryIPLastRefreshTime = time.Now()
+	res, err := c.getSecondaryIPs(ctx)
+	return true, res, err
+}
+
+func (c *Client) getSecondaryIPs(ctx context.Context) ([]string, error) {
+	req, err := c.buildRequest(ctx, &GetSecondaryIPsRequest{})
+	if err != nil {
+		return nil, errors.Wrap(err, "building request")
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "submitting request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, die(resp.StatusCode, resp.Header, resp.Body, req.URL.Path)
+	}
+
+	return parseSecondaryIPsFromWireServerResponse(resp)
+}
+
+func parseSecondaryIPsFromWireServerResponse(resp *http.Response) (res []string, err error) {
```

Comment:
This should generally parse the response and put it in a set of structs (in responses.go). Drilling into the response should be done by the user of the client, not the client itself. This allows it to be a general client for NMAgent, not a one-off data grabber (which was what this package was intended to replace).